### ISSUE TITLE
Output Feature Point Score Values

### ISSUE THREAD

**Issue by abhilashjain7** [rpautrat/SuperPoint#106]
Hi! Thank you for making this amazing code available!
There's an issue I encountered which I cannot find listed here. I retrained the entire SuperPoint network (with default configurations) and found that the score (or response) value of the feature points was significantly lower compared to what I get from the pretrained network (downloaded from [here](https://github.com/MagicLeapResearch/SuperPointPretrainedNetwork)). The maximum score value from the pretrained network lie in the range 0.7-0.8, whereas the maximum value from my own trained network ranges in between 0.4-0.5 (on the same image).
Can you please explain why this is happening and how to rectify this?

**Comment by rpautrat** [comment]
Hi!
Even though I tried to follow the original implementation as accurately as possible, it is still possible that this implementation doesn't match perfectly the original one.
Your remark is interesting and could be explain by a difference in the parameters used during training. In particular the number of keypoints kept in the generation of the pseudo ground truth and the detection threshold might play a role in the max value of the detection: if you train your network to recognize less keypoints, the few detections will be more peaky and close to 1, whereas if you train it with multiple keypoints, the score map will be more spread.

**Comment by abhilashjain7** [comment]
Yes, I thought so too.
I noticed one more thing, that while exporting MS-COCO pseudo-ground truth points, the code only dumps the coordinates of the points, and no score values. This means that the score **label** of these points is always 1, no matter the actual "strength" of the point. Essentially, the network doesn't learn the difference between a "strong" and "weak" keypoint. Would it help if along with the keypoint coordinates, we also used the score values as labels during training on MS-COCO?

**Comment by rpautrat** [comment]
It would be possible, but I fear that it would spread even more the score values. With our current entropy loss it actually encourages to get peaks of scores: it is either a keypoint or not at all.

But I agree that if you want to vary the detection threshold later, it would be better to train directly with ground truth score maps so that your ground truth is not directly dependent on the threshold you used when exporting the pseudo ground truth.

**Comment by abhilashjain7** [comment]
Ok. I will try and experiment with that..
Thank you!
